### ISSUE TITLE
Support `sourceRoot` from `SourceMapBuilder`

### ISSUE THREAD

**Issue by kdy1** [getsentry/rust-sourcemap#50]
Related: https://github.com/swc-project/swc/issues/2362

I want to generate sourcemap like

```json
{
  "version": 3,
  "sourceRoot": "/Users/pbadenski/workspace/project/",
  "sources": [
    "test.ts"
  ],
}
```

but I found that `SourceMapBuilder` does not provide a way to generate a sourcemap with `sourceRoot`

**Comment by kamilogorek** [comment]
Processed `SourceMap` struct, has `sourceRoot` already resolved, and all `sources` are correctly mapped to appropriate paths.

We could potentially add `extract_source_root` option to `RewriteOptions`, which could then be used like so:

```rust
let sm = SourceMap::from_reader(input)?;
let processed_sm = sm.rewrite(&RewriteOptions {
    extract_source_root: true,
    ..Default::default()
})?;
processed_sm.to_writer(output)?;
```

wdyt @Swatinem @loewenheim?

**Comment by kdy1** [comment]
Thank you!

**Comment by kamilogorek** [comment]
Published in `6.2.0`.
